### PR TITLE
Build: add sanitizer stages

### PR DESCRIPTION
**Pull request by pstorz**
This PR adds the option to run the bareos tests with sanitizers enabled ("-D ENABLE_SANITIZERS=yes") to immediately detect memory problems that the sanitzers can detect.

It also introduces the test label `broken_with_enabled_sanitzers` which marks tests that do not work with sanitizers enabled. These tests are skipped in the sanitizer stage.

It also fixes a memleak that was detected by the sanitizers stage in `mntent_cache.cc`

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Separate commit for this PR in the CHANGELOG.md, PR number referenced is same
- [x] Commit descriptions are understandable and well formatted
- [x] If backport: add original PR number and target branch at top of this file: **Backport of PR#000 to bareos-2x**

### CODE REVIEW

File: .matrix.yml [hunk 1]
```diff
@@ -3,6 +3,44 @@ JENKINS_TESTTYPES:
   - univention-bareos
 
 OS:
+  Sanitize:
+    "F37":
+      TYPE: scripted
+      IMAGE: fedora37
+      BUILD_SCRIPT: CD/sanitize/build-sanitize.sh
```
Comment (arogge) [inline .matrix.yml:10]: we can move that to the Bareos repository, you just use "bareos/..." as a path here.
The script path is relative to Jenkins' $WORKSPACE and it always run with cwd set to the top of the cloned bareos repository.
So you can test and develop the script locally and it should just work in Jenkins.

File: .matrix.yml [hunk 2]
```diff
@@ -3,6 +3,44 @@ JENKINS_TESTTYPES:
...
+      FINISH_SCRIPT: CD/sanitize/finish-sanitize.sh
+      BUILD_FOR_VERSIONS:
+        - bareos-22
```
Comment (arogge) [inline .matrix.yml:13]:
```suggestion
```
This is not used anymore if the matrix.yml is in the Bareos repository.

File: .matrix.yml [hunk 3]
```diff
@@ -3,6 +3,44 @@ JENKINS_TESTTYPES:
...
+      ARCH:
+        - x86_64
+      PROJECTPACKAGES:
+        x86_64:
+          - bareos
+
```
Comment (arogge) [inline .matrix.yml:43]:
```suggestion
```
One single stage will probably be sufficient.

File: devtools/dist-tarball.sh [hunk 4]
```diff
@@ -123,7 +123,7 @@ fi
 (echo -ne 'cmake/BareosVersion.cmake\0'; "$git" ls-files -z) | \
 "$sort" -u -z | \
 "$tar" "${args[@]}" -cf - --files-from - | \
-"$xz" -c -6 > "${archive_file}"
+"$xz" --threads=4 -c -6 > "${archive_file}"
```
Comment (arogge) [inline devtools/dist-tarball.sh:126]:
```suggestion
"$xz" --threads=0 -c -6 > "${archive_file}"
```
Setting threads=0 will use all available cpu-cores.

File: .matrix.yml (hunk 2 above)
Comment (pstorz) [reply .matrix.yml]: Done

File: .matrix.yml (hunk 1 above)
Comment (pstorz) [reply .matrix.yml]: done

**Review by pstorz — COMMENTED** [review]
Schaut doch gut aus!

**Review by arogge — APPROVED** [review]

**Pull request merged**